### PR TITLE
feat(onboarding): Present Terms to users upgrading from v1 or those who need to accept updated Terms

### LINKED ISSUE #21113: Show Privacy Policy and Terms modal for upgraded users
When users upgrade from V1 to V2 app they need to accept updated Privacy Policy and Terms. 

For those purposes we need to show Privacy Policy modal after upgrade. 

**Designs**

https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=13865-45279&t=YzMcbvPYvKIJ31NR-1

image

### PR DESCRIPTION
**Pull request by ilmotta**
Fixes https://github.com/status-im/status-mobile/issues/21113

### Summary

This PR is 99% identical to https://github.com/status-im/status-mobile/pull/21124. **We are cherry-picking** from the 2.30 release branch revision https://github.com/status-im/status-mobile/commit/d45eb5ec20ae7757cd99554920e466eabac0bfb1 and applying to the status-go and status-mobile `develop` branches in preparation for the 2.31 release.

Related status-go PR: https://github.com/status-im/status-go/pull/5977

We will update the release branches once the develop branches are merged.

### Steps to test

Steps to test are identical to the original PR https://github.com/status-im/status-mobile/pull/21124.

I re-tested and everything seems to work as well as in 2.30, where the original PR was already tested. I tested a scenario that's difficult for a QA to test, where I simulated a user who had Status v2 installed, had already accepted the Terms, then I forced a new migration in status-go to simulate an update to the Terms, and finally checked the user would need to re-accept terms before proceeding.

status: ready

### CODE REVIEW

File: src/status_im/subs/profile.cljs [hunk 1]
```diff
@@ -17,6 +18,15 @@
  (fn [{:keys [customization-color]}]
    (or customization-color constants/profile-default-color)))
 
+;; A profile can only be created without accepting terms in Status v1. In Status
```
Comment (ilmotta) [inline src/status_im/subs/profile.cljs:21]: I was confused by my own comment in the original commit, so after cherry-picking I reworded things a bit and renamed the subscription.

**Review by qfrank — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ilmotta thanks for the PR. Please take a look at the issue below

### ISSUE 1 Terms screen appears after removing one of the existing Profiles despite terms have been already accepted before

**Steps:** 

1. Create Profile A
2. Close and reopen the app 
3. Create Profile B
4. Close and reopen the app
5. Go to Profiles list screen
6. Remove Profile A
7. Close and reopen the app
8. Observe which screen is displayed after app reopening

**Actual result:** Terms screen is displayed despite terms have been already accepted. 


https://github.com/user-attachments/assets/ccb2a061-d4a6-4026-adcc-e174e959f494

**Comment by ilmotta** [comment]
Thank you for promptly testing @pavloburykh 💯 

Unfortunately this is the behavior by "design" in the code. It's a known shortcoming of the implementation in status-go. The state about whether the user accepted the terms is stored in the first profile ever created. If you created two profiles, but deleted the first one, then the incorrect screen will appear. If you remove profile B instead of A, it work correctly because the state would still be stored in the local db by status-go.

In release 2.32 we will make same pretty significant simplifications to onboarding. One of the changes under consideration is to remove this mandatory checkbox and use something like "By proceeding you agree with the terms" and possibly removing this extra screen saying "Explore new Status" because most users would have migrated from v1 anyway (according to app stores). It will be a matter of aligning with legal too, so let's see. Right now we can minimize our efforts regarding the privacy policy and terms of use UX because we may change everything soon.

For now I would say this is a minor limitation we can live with until 2.32 is published.

What do you think?

**Comment by pavloburykh** [comment]
> What do you think?

Agree @ilmotta! PR is ready for merge. Thank you!

**Comment by pavloburykh** [comment]
Hey @ilmotta! Also, it turned out that Terms screen is appearing for synced account. 

### ISSUE 2 Terms screen appears after syncing the account

**Steps:** 

1. Generate sync QR on Device A
2. Scan QR by Device B to sync the devices
3. Wait until sync is complete
4. Close and reopen the app on Device B

**Actual result:** user needs accept Terms


https://github.com/user-attachments/assets/32d179c1-c0f6-4746-8d71-ec894e5a5810

**Comment by pavloburykh** [comment]
@ilmotta how do you think, do we want to fix [ISSUE 1](https://github.com/status-im/status-mobile/pull/21487#issuecomment-2431934024) and [ISSUE 2](https://github.com/status-im/status-mobile/pull/21487#issuecomment-2455062945) or we will leave it as it is. In case we want to fix it, I can combine and log them in one separate issue.

**Comment by ilmotta** [comment]
> @ilmotta how do you think, do we want to fix [ISSUE 1](https://github.com/status-im/status-mobile/pull/21487#issuecomment-2431934024) and [ISSUE 2](https://github.com/status-im/status-mobile/pull/21487#issuecomment-2455062945) or we will leave it as it is. In case we want to fix it, I can combine and log them in one separate issue.

@pavloburykh, `ISSUE 2` is new to me, and it's an unfortunate bug we will have to live with in 2.31 (the bug should exist in 2.30 (?) because we basically cherry-picked).

But to answer your question, onboarding simplifications have been defined and the user won't have to accept the checkbox (per screenshot). This means the existing (2.30 & 2.31) solution won't work well in 2.32 and we'll need to adjust. The business problem is: how do we tell the user the Terms of Service or Privacy Policy changed between releases? I believe whatever we do we will do in the least obtrusive way and the simplest way, therefore different than 2.30 & 2.31.

Long story short, I think we shouldn't fix ISSUE 1 or ISSUE 2 now, but they will be attacked/solved in 2.32 once we define & implement what I shared above. I even have the impression we might completely remove the code from this PR (and from status-go), but let's see.

**Comment by pavloburykh** [comment]
> (the bug should exist in 2.30 (?) 

Yes. it exists in 2.30, wondering how we (QAs) have missed it. 
Thanks for the answer @ilmotta Then, let's wait for implementing onboarding simplification.

**Comment by ilmotta** [comment]
> > (the bug should exist in 2.30 (?)
> 
> Yes. it exists in 2.30, wondering how we (QAs) have missed it. Thanks for the answer @ilmotta Then, let's wait for implementing onboarding simplification.

@pavloburykh I didn't forget about this issue :) But in 2.32 we will go with what we have right now and not cause further modifications because the onboarding simplifications are still in progress. In release 2.33 we will likely have time to get back to this issue. My intuition tells me we will just remove the entire feature because users of 2.32+ will accept the terms implicitly.
